bin/node/src/chain_spec.rs

```diff
@@ -201,38 +201,26 @@ fn system_properties(token_symbol: String) -> serde_json::map::Map<String, Value
     .collect()
 }
 
-/// Generate chain spec.
-pub fn config(
-    chain_params: ChainParams,
-    authorities: Vec<AuthorityKeys>,
-) -> Result<ChainSpec, String> {
-    generate_chain_spec_config(chain_params, authorities, None, None)
-}
-
 /// Generate chain spec for local runs.
-/// Stash and controller accounts are generated for the specified authorities.
+/// Controller accounts are generated for the specified authorities.
 pub fn local_config(
     chain_params: ChainParams,
     authorities: Vec<AuthorityKeys>,
 ) -> Result<ChainSpec, String> {
-    let authority_accounts: Vec<AccountId> = to_account_ids(&authorities).collect();
-    let controller_accounts: Vec<AccountId> =
-        generate_staking_accounts(authority_accounts.clone(), StakingAccount::Controller);
-    let stash_accounts: Vec<AccountId> =
-        generate_staking_accounts(authority_accounts, StakingAccount::Stash);
-    generate_chain_spec_config(
-        chain_params,
-        authorities,
-        Some(controller_accounts),
-        Some(stash_accounts),
-    )
+    let controller_accounts: Vec<AccountId> = to_account_ids(&authorities)
+        .into_iter()
+        .enumerate()
+        .map(|(index, _account)| {
+            get_account_id_from_seed::<sr25519::Public>(format!("//{}//Controller", index).as_str())
```

Comment:
out of curiosity (I have never worked with this code before):
 - why do we use different derivation method than in `aleph-client`? I mean `get_account_id_from_seed`. From what I can see, here we hide under some traits, apparently trying to be generic, but actually this is the same code as in `aleph-client`. Maybe we should stick to one method?
 - why don't we make a use of `_account` for controller (previously also stash) accounts? is this for convenience? seems suspicious

Reply:
Actually, this seems to be a pretty important point. @Marcin-Radecki reworked this from my previous commit which used this derivation, which in turn came from convenience functions for chain spec generation. Right now, with more knowledge about e2e tests, it seems that using `get_account_id_from_seed` will probably bite us. You may correct me if I am wrong, but this will result in different keys than using `keypair_from_string` from `aleph-client`. So, I would propose either to go with something based on `kaypair_from_string` or something which gives us the same keys we produce in the bench performance test (e.g. https://github.com/Cardinal-Cryptography/aleph-node/blob/3f72b3fe25ea433cdaba954f8d1c0f0c2f5236a8/benches/payout-stakers/src/main.rs#L73).

Reply:
Might go slightly further than that - we probably want to make sure that all the key pairs and accounts based on them are consistent with `alpeh-client` and `payout-stakers`, not only the lines @pmikolajczyk41 initially mentioned.

Reply:
Good points, let's use `keypair_from_string` everywhere we use now `get_account_id_from_seed`

Reply:
@pmikolajczyk41  On the other hand, this implies adding `aleph-client` dependency to `aleph-node`. Not sure if it's worth it, to decrease one-liner duplication we'd need to build 395 packages more etc. There's another argument, this time about architecture - `aleph-client` was intended to use with non-production code crates. One can argue this does not make any sense since every code is a production code, test code or not, however, in practice, only test-code crates use `aleph-client` for now. So I'll stick to `get_account_id_from_seed`  for now.

Reply:
@maciejzelaszczyk how's exactly get_account_id_from_seed going to give us different keys than keypair_from_string ?

Reply:
nonono, no way of depending on aleph-client; however, iirc `keypair_from_string` is a very short code which does almost the same as the snippet here - I would just copy it to remove some unnecessary abstractions from here

Reply:
Okay that's better idea, let me do it